XYZDrivers/src/main/java/com/xyzdrivers/repositories/ClaimsRepo.java

```diff
@@ -78,16 +78,26 @@ public void insert(Claim c) throws RepositoryException {
                     throw new IllegalArgumentException("One of the declared fields in object c is null.");
                 }
             }
-                                      
-            splitClaim[0] = c.getMemberID();
-            splitClaim[1] = new Date(c.getDate().getTimeInMillis());
-            splitClaim[2] = c.getReason();
-            splitClaim[3] = c.getStatus();
-            splitClaim[4] = c.getAmount();
             
-            String columns = "(MEM_ID, DATE, RATIONALE, STATUS, AMOUNT)";
-        
-
+            // Generate a unique ID from existing row count
+            int nextFreePrimaryKey = sqlService.getNextFreeUniqueInteger("CLAIMS", "id");
```

Comment:
I don't understand the need for this - the table id automatically increases. If we're going to do this then we might as well remove the ability for tables to auto increment. And develop does not throw an exception when a claim is submitted.

Reply:
Agreed, you should never manually set an id, especially when auto incrementing